lib/jsdom/living/crypto/Crypto-impl.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+const nodeCrypto = require("crypto");
+const DOMException = require("domexception/webidl2js-wrapper");
+
+// https://w3c.github.io/webcrypto/#crypto-interface
+class CryptoImpl {
+  constructor(globalObject) {
+    this._globalObject = globalObject;
+  }
+
+  // https://w3c.github.io/webcrypto/#Crypto-method-getRandomValues
+  getRandomValues(array) {
+    // Note: this rejects Float32Array and Float64Array.
+    if (!(array instanceof this._globalObject.Int8Array ||
+        array instanceof this._globalObject.Uint8Array ||
+        array instanceof this._globalObject.Uint8ClampedArray ||
+        array instanceof this._globalObject.Int16Array ||
+        array instanceof this._globalObject.Uint16Array ||
+        array instanceof this._globalObject.Int32Array ||
+        array instanceof this._globalObject.Uint32Array ||
+        array instanceof this._globalObject.BigInt64Array ||
+        array instanceof this._globalObject.BigUint64Array)) {
```

Comment:
This will fail on cross-realm instances, which is not really what the spec intends (though the spec is not precise).

The best way I can think of to do this is to check `array.constructor.name`.

Reply:
I must admit that I'm completely out of my depth when it comes to realms. I used `array.constructor.name` as suggested.